src/test/java/tools/jackson/databind/ext/javatime/ser/WriteDatesAsTimestamps5142JavaTimeTests.java

```diff
@@ -0,0 +1,160 @@
+package tools.jackson.databind.ext.javatime.ser;
+
+import java.time.*;
+
+import org.junit.jupiter.api.Test;
+import tools.jackson.databind.ObjectMapper;
+import tools.jackson.databind.cfg.DateTimeFeature;
+import tools.jackson.databind.ext.javatime.DateTimeTestBase;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+// [databind#5142] JSTEP-5 Tests to verify current behavior and discuss further action DateTimeFeature.WRITE_DATES_AS_TIMESTAMPS
+public class WriteDatesAsTimestamps5142JavaTimeTests
+    extends DateTimeTestBase
+{
+    private static final ObjectMapper WITH_TIMESTAMP_MAPPER = mapperBuilder()
+            .enable(DateTimeFeature.WRITE_DATES_AS_TIMESTAMPS)
+            .build();
+
+    private static final ObjectMapper WITHOUT_TIMESTAMP_MAPPER = mapperBuilder()
+            .disable(DateTimeFeature.WRITE_DATES_AS_TIMESTAMPS)
+            .build();
+
+    @Test
+    public void testWriteDatesAsTimeStamps() throws Exception {
+        // java.time.OffsetDateTime
+        _testTimestamp(
+                LocalDateTime.of(2025, 5, 4, 18, 1, 0),
+                LocalDateTime.class,
+                // Expected "[2025,5,4,18,1,0]",
+                "[2025,5,4,18,1]", // Acutal
+                "\"2025-05-04T18:01:00\""
+        );
+        // java.time.ZonedDateTime
+        _testTimestamp(
+                LocalDateTime.of(2025, 5, 4, 18, 1, 2),
+                LocalDateTime.class,
+                "[2025,5,4,18,1,2]",
+                "\"2025-05-04T18:01:02\""
+        );
+        // java.time.LocalDate
+        _testTimestamp(
+                LocalDate.of(2025, 5, 4),
+                LocalDate.class,
+                "[2025,5,4]",
+                "\"2025-05-04\""
+        );
+        // java.time.LocalTime
+        _testTimestamp(
+                LocalTime.of(18, 1, 2),
+                LocalTime.class,
+                "[18,1,2]",
+                "\"18:01:02\""
+        );
+        // java.time.Instant
+        _testTimestamp(
+                Instant.ofEpochMilli(1234567890123L),
+                Instant.class,
+                // Expected "1234567890123",
+                "1234567890.123000000",
```

Comment:
This depends on setting for `DateTimeFeature.WRITE_DATE_TIMESTAMPS_AS_NANOSECONDS`, I think.

We need to solve the question of "millis-or-nanos" (Joda, java.util only do milliseconds; Java 8 time supports nanosecond precision).